code/modules/atmospherics/machinery/components/unary_devices/thermomachine.dm

```diff
@@ -200,6 +221,17 @@
 	use_power(power_usage)
 	update_parents()
 
+/obj/machinery/atmospherics/components/unary/thermomachine/attackby(obj/item/assembly/signaler/anomaly/pyro, mob/living/user, params)
```

Comment:
You never actually check for anomaly type to make sure it's a pyro one, and it doesn't check for type at ALL, doesn't ANY item in the game work to be set as anomaly core?

Reply:
Yeah... I didn't know you were supposed to check it and assumed it worked by checking the item in attackby, fixed.